onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -2290,6 +2290,83 @@ It's an extension of Gelu. It takes the sum of input A and bias input B as the i
         }
       });
 
+  static const char* TorchEmbedding_ver1_doc = R"DOC(
+      Based on Torch operator Embedding, creates a lookup table of embedding vectors of fixed size,
+       for a dictionary of fixed size.
+      )DOC";
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(TorchEmbedding)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .SetDoc(TorchEmbedding_ver1_doc)
+      .Input(
+          0,
+          "weight",
+          "The embedding matrix of size N x M. 'N' is equal to the maximum possible index + 1, and 'M' is "
+          "equal to the embedding size",
+          "T")
+      .Input(
+          1,
+          "indices",
+          "Long tensor containing the indices to extract from embedding matrix.",
+          "tensor(int64)")
+      .Input(
+          2,
+          "padding_idx",
+          "A 0-D scalar tensor. If specified, the entries at `padding_idx` do not contribute to the gradient; "
+          "therefore, the embedding vector at `padding_idx` is not updated during training, ",
+          "i.e. it remains as a fixed pad."
+          "tensor(int64)",
+          OpSchema::Optional)
+      .Input(
+          3,
+          "scale_grad_by_freq",
+          "A 0-D bool tensor. If given, this will scale gradients by the inverse of frequency of "
+          "the indices (words) in the mini-batch. Default  is ``False``",
+          "tensor(bool)",
+          OpSchema::Optional)
```

Comment:
Comparing with torch.nn.functional.embedding, we are missing a few inputs, particularly max_norm and norm_type. Will the exporter take care of adding the appropriate nodes downstream to this op for normalization if these are provided?

Reply:
These arguments are present in at python level, but the backend op does not receive these arguments.
Here is the ATen op implementation: https://github.com/pytorch/pytorch/blob/a28c7db9f906a3dd445094e670b9b80b2feec9ec/aten/src/ATen/native/Embedding.cpp#L16

So, when converting the python module, the corresponding ops will be added.

Reply:
Thanks for clarifying this, Negin! Your reply prompted me to look into the implementation of `max_norm`:

https://github.com/pytorch/pytorch/blob/645119eaefd0dcf1afc54e9ad58678b5245dea78/torch/nn/functional.py#L1960-L1962

I don't think the right thing will happen if we try to export with `max_norm` set to anything other than `None` - even after adding the contrib op - because the definition relies on modifying the embedding matrix in place, and I don't think we have a way to represent that in the ONNX graph. 

Not a blocker for this PR, but do you think there is any way we could raise an error when we hit this case? (/cc @spandantiwari for extra eyes on the problem).

Reply:
So, the op which is used to normalize the Embedding layer is an ATen op called embedding_renorm_.
It is correct that this is an inplace op, but exporter does have some preprocessing to take care of inplace ops.
However, we currently do not have export support for this op. As you mentioned this is outside of the scope of this PR. But we can support for this op in exporter in general. No blockers.

Reply:
Right... I think there might still be a blocker because IIUC the exporter won't export a graph that has a side effect on one of its inputs (i.e. the embedding weights), so the visible side effect of running the exported graph as an ORTModule would be different from running the original module.